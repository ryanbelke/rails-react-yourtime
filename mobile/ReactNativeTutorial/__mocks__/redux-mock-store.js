import configureMockStore from 'redux-mock-store';
import { initialState as reduxInitialState } from 'ReactNativeTutorial/app/reducers';

import mockThunkMiddleware from './mockThunkMiddleware';

export const createStoreFromState = configureMockStore([mockThunkMiddleware]);
export const initialState = reduxInitialState;

export default () => createStoreFromState(initialState);
