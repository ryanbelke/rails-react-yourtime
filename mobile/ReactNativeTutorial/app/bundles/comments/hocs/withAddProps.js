// @flow
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import commentFormSelector from 'ReactNativeTutorial/app/selectors/commentFormSelector';
import * as actions from '../thunks';

export type AddPropsType = {
  author?: string,
  text?: string,
  actions: {
    fetch: () => void,
    updateForm: (payload: Object) => void,
    createComment: (payload: Object) => void,
  }
}

const mapStateToProps = createSelector(
  commentFormSelector,
  (commentForm: any) => commentForm.toJS()
);

const mapDispatchToProps = (dispatch: Function) => ({
  actions: bindActionCreators(actions, dispatch),
});

export default (Component: ReactClass<AddPropsType>) => connect(
  mapStateToProps,
  mapDispatchToProps
)(Component);
