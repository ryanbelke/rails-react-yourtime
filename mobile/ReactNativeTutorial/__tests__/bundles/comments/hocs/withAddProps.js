import React from 'react';
import withAddProps from 'ReactNativeTutorial/app/bundles/comments/hocs/withAddProps';
import Mock from 'mock';
import renderer from 'react-test-renderer';
import {createStoreFromState, initialState} from 'redux-mock-store';

describe('withAddProps', () => {
  it('adds AddProps to a component', () => {
    const state = initialState.mergeDeep({
      commentForm: {
        author: 'Alexey',
        text: 'Random comment',
      },
    });
    const store = createStoreFromState(state);
    const Component = withAddProps(Mock);
    const tree = renderer.create(<Component store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
