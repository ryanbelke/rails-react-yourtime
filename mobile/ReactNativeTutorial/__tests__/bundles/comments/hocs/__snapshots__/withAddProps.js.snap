exports[`withAddProps adds AddProps to a component 1`] = `
<Mock
  actions={
    Object {
      "createComment": [Function],
      "fetch": [Function],
      "updateForm": [Function],
    }
  }
  author="Alexey"
  meta={Object {}}
  text="Random comment" />
`;
